File: Day 20 - Sorting/solution.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', function(inputStdin) {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.split('\n');

    main();
});

function readLine() {
    return inputString[currentLine++];
}



function main() {
    var n = parseInt(readLine());
    var a = readLine().split(' ');
    a = a.map(Number);

    var swapped;
    var numSwaps = 0;
    do{
        swapped = false;
        for(var i = 0; i < n - 1; i++){
            if(a[i] > a[i + 1]){
                var temp = a[i];
                a[i] = a[i + 1];
                a[i + 1] = temp;
                swapped = true;
                numSwaps++
            }
        }
    }while(swapped);
    console.log('Array is sorted in ' + numSwaps + ' swaps.');
    console.log('First Element:',a[0]); 
    console.log('Last Element:',a[n - 1]);
}

File: Day 20 - Sorting/Solution.java
import java.util.Scanner;

public class Solution {
    
    public static void main(String[] args){
        
        Scanner input = new Scanner(System.in);
        int n = input.nextInt();
        int array[] = new int[n];
        for(int i = 0; i < n; i++){
            array[i] = input.nextInt();
        }

        int numSwaps = 0;
        for(int i = 0; i < n; i++){
            for(int j = 0; j < n - 1; j++){
                if(array[j] > array[j+1]){
                    int tmp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = tmp;
                    numSwaps++;
                }
            }

            if(numSwaps == 0){
                break;
            }
        }

        System.out.println("Array is sorted in " + numSwaps + " swaps.");
        System.out.println("First Element: "+ array[0]);
        System.out.println("Last Element: " + array[array.length - 1]);
    }
}
